app/src/androidTest/java/com/lightricks/feedexercise/data/FeedRepositoryTest.kt

```diff
@@ -1,15 +1,76 @@
 package com.lightricks.feedexercise.data
 
+import android.content.Context
+import androidx.arch.core.executor.testing.InstantTaskExecutorRule
 import androidx.lifecycle.LiveData
 import androidx.lifecycle.Observer
+import androidx.room.Room
+import androidx.test.core.app.ApplicationProvider
 import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.lightricks.feedexercise.database.FeedDatabase
+import com.lightricks.feedexercise.database.FeedItemDao
+import com.lightricks.feedexercise.database.FeedItemEntity
+import com.lightricks.feedexercise.network.MockFeedApiService
+import junit.framework.Assert.assertEquals
+import org.hamcrest.Matchers.greaterThan
+import org.junit.After
+import org.junit.Assert.assertThat
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
 import org.junit.runner.RunWith
+import java.io.IOException
 import java.util.concurrent.CountDownLatch
 import java.util.concurrent.TimeUnit
 
 @RunWith(AndroidJUnit4::class)
 class FeedRepositoryTest {
-   //todo: add the tests here
+
+    private lateinit var feedDao: FeedItemDao
+    private lateinit var db: FeedDatabase
+    private lateinit var mockFeedApiService: MockFeedApiService
+
+    @Before
+    fun createDependencies() {
+        val context = ApplicationProvider.getApplicationContext<Context>()
+        db = Room.inMemoryDatabaseBuilder(
+            context, FeedDatabase::class.java
+        ).build()
+        feedDao = db.feedItemDao()
+        mockFeedApiService = MockFeedApiService(context)
+    }
+
+    @After
+    @Throws(IOException::class)
+    fun closeDb() {
+        db.close()
+    }
+
+    @Rule
+    @JvmField
+    val rule = InstantTaskExecutorRule()
+
+    @Test
+    fun writeToDBReadFeedItems() {
+        val feedItemEntities = mutableListOf<FeedItemEntity>()
+        feedItemEntities.add(FeedItemEntity("1", "temp", true))
+        val repository = FeedRepository(mockFeedApiService, db)
+        feedDao.insertAll(feedItemEntities).blockingAwait()
+        val res = repository.getFeedItems().blockingObserve()
+        if (res != null) {
+            assertEquals(feedItemEntities.size, res.size)
```

Comment:
If you write just one entity, you could as well compare it's properties (ID, URL, etc.)